Fix a few text related issues

applenick opened the pull request:
# Fix a few text related issues
Most of these issues come from the recent text refactor, all of which are resolved by this PR 🎉 

* Vanished mapmakers are no longer exposed via various map info locations
* Fixed issue related to names not rendering decorations (death message bug)
* Resolve team chat prefix not using proper aliased name (resolves #534)

## Screenshot
![Screen Shot 2020-06-14 at 1 11 40 AM](https://user-images.githubusercontent.com/3377659/84588529-0dda2800-addd-11ea-8cd1-48a2ebe20583.png)

![Screen Shot 2020-06-14 at 12 34 22 AM](https://user-images.githubusercontent.com/3377659/84588535-13377280-addd-11ea-89fd-544bf6f6edfa.png)

Signed-off-by: applenick <applenick@users.noreply.github.com>

Electroid commented on core/src/main/java/tc/oc/pgm/teams/Team.java:
```diff
@@ -37,8 +37,6 @@
   private TeamMatchModule tmm;
   private JoinMatchModule jmm;
   protected @Nullable String name = null;
-  protected @Nullable Component componentName;
```
We should be caching the `Componet` name. Instead, listen to the team rename event and change it.

Electroid commented on util/src/main/java/tc/oc/pgm/util/text/types/PlayerComponent.java:
```diff
@@ -38,32 +38,49 @@ static Component of(@Nullable Player player, String defName, NameStyle style) {
 
-    Component usernameComponent;
+    // For name styles that don't allow vanished, make vanished appear offline
+    if (!style.showVanish && isVanished(player)) {
```
I think we need to clean up this code. Maybe separate methods for each `NameStyle`?

applenick replied:
I agree, with the previous refactor I tried to keep it similar to the old `PersonalizedPlayer` class, but imo would be better to separate into different methods as you suggested 👍 

I’ll take care of this in a bit.

applenick commented:
@Electroid 
I reverted the change related to Team components caching. I decided against using the team rename event and instead opted for a reset to the chatPrefix which should fix the issue with team names updating.

As for `PlayerComponent`, I went ahead and cleaned up `NameStyle` to be more aligned with recent changes and what’s currently supported. Also as requested I split up the rendering for each style into different methods. 

Let me know if you have any notes or suggestions! Tested and everything works so should be fine 👍

Electroid commented on util/src/main/java/tc/oc/pgm/util/text/types/PlayerComponent.java:
```diff
@@ -81,6 +68,109 @@ static Component of(CommandSender sender, NameStyle style) {
   }
 
+  // What an offline or vanished username renders as
+  static Component formatOffline(String name, boolean plain) {
```
Make all of these functions return `Component.Builder` to avoid copying `Component`s among the methods. In `PlayerComponent.of` you can build it before returning it.

applenick replied:
Ah better idea! I’ll fix this up 👍

applenick replied:
I believe you meant `TextComponent.Builder`, but it’s all cleaned up now 👍
